server/src/main/java/io/druid/segment/loading/LocalDataSegmentPusher.java

```diff
@@ -54,7 +54,7 @@ public LocalDataSegmentPusher(
   @Override
   public String getPathForHadoop(String dataSource)
   {
-    return String.format("file://%s/%s", config.getStorageDirectory(), dataSource);
+    return String.format("file://%s/%s", config.getStorageDirectory().getAbsoluteFile(), dataSource);
```

Comment:
is it at all possible to use standard URI building things here? Either hadoop FS path building or the java File and Path constructs?